eng/Versions.props

```diff
@@ -25,13 +25,23 @@
   </PropertyGroup>
   <!-- Production Dependencies -->
   <PropertyGroup>
+    <SystemCollectionsImmutableVersion>6.0.0</SystemCollectionsImmutableVersion>
```

Comment:
We weren't consistent about it before so no concerns about you not doing it but I moved these to the "Production Dependencies" group rather than "toolset" (except for the one that's actually toolset).